### PR TITLE
PE-895: Implement methods for rename-folder ArFSDAO

### PR DESCRIPTION
**Pull request by matibat**

### CODE REVIEW

File: src/arfs/arfsdao.ts [hunk 1]
```diff
@@ -1537,6 +1543,80 @@ export class ArFSDAO extends ArFSDAOAnonymous {
 		};
 	}
 
+	async renamePublicFolder({
+		folder,
+		newName,
+		metadataRewardSettings
+	}: ArFSRenamePublicFolderParams): Promise<ArFSRenamePublicFolderResult> {
+		// Prepare meta data transaction
+		const metadataTxData = new ArFSPublicFileMetadataTransactionData(
```
[the remark this answers is not in the document]
Comment (arielmelendez) [reply src/arfs/arfsdao.ts:1552]: Basically, I think we should augment `ArFSPublicFolderTransactionData` to allow us to create them from an existing set of folder data. Also, we shouldn't use File things in places where we're doing Folder things.

[the remark this answers is not in the document]
Comment (matibat) [reply src/arfs/arfsdao.ts:1552]: This is the result of Copy-Pasta :spaghetti:

[the remark this answers is not in the document]
Comment (matibat) [reply src/arfs/arfsdao.ts]: Let's see here: 7789c11

**Review by arielmelendez — APPROVED** [review]
Looks good. My only suggestion would be to D.R.Y. the code out using a private function that takes closure params and generics to handle the type differences. See moveEntity for inspiration. This is fine for now though.

**Review by fedellen — APPROVED** [review]

**Comment by matibat** [comment]
Just created a tech-debt ticket (PE-936) to D.R.Y. out the code

**Pull request merged**